### ISSUE TITLE
Flux bootstrap stuck at deployment "source-controller" successfully rolled out

### ISSUE THREAD

**Issue by binhnguyenduc** [fluxcd/flux2#767]
Hello, I'm trying to install flux2 to migrate my existing flux1 over. However, the installation seems to be stuck at `deployment "source-controller" successfully rolled out`, no matter how many I run the bootstrap command. All the deployments are green and Active. 

We are running Kubernetes 1.18.3 with Rancher.

Here is my command:
`flux bootstrap gitlab --components-extra=image-reflector-controller,image-automation-controller --ssh-hostname=gitlab.com --owner=*** --repository=*** --branch=dev --path=/clusters/uat --verbose --log-level=debug`

```
► connecting to gitlab.com
✔ repository cloned
✚ generating manifests
✔ components are up to date
► installing components in flux-system namespace
namespace/flux-system unchanged
customresourcedefinition.apiextensions.k8s.io/alerts.notification.toolkit.fluxcd.io configured
customresourcedefinition.apiextensions.k8s.io/buckets.source.toolkit.fluxcd.io configured
customresourcedefinition.apiextensions.k8s.io/gitrepositories.source.toolkit.fluxcd.io configured
customresourcedefinition.apiextensions.k8s.io/helmcharts.source.toolkit.fluxcd.io configured
customresourcedefinition.apiextensions.k8s.io/helmreleases.helm.toolkit.fluxcd.io configured
customresourcedefinition.apiextensions.k8s.io/helmrepositories.source.toolkit.fluxcd.io configured
customresourcedefinition.apiextensions.k8s.io/imagepolicies.image.toolkit.fluxcd.io configured
customresourcedefinition.apiextensions.k8s.io/imagerepositories.image.toolkit.fluxcd.io configured
customresourcedefinition.apiextensions.k8s.io/imageupdateautomations.image.toolkit.fluxcd.io configured
customresourcedefinition.apiextensions.k8s.io/kustomizations.kustomize.toolkit.fluxcd.io configured
customresourcedefinition.apiextensions.k8s.io/providers.notification.toolkit.fluxcd.io configured
customresourcedefinition.apiextensions.k8s.io/receivers.notification.toolkit.fluxcd.io configured
role.rbac.authorization.k8s.io/crd-controller-flux-system unchanged
rolebinding.rbac.authorization.k8s.io/crd-controller-flux-system unchanged
clusterrolebinding.rbac.authorization.k8s.io/cluster-reconciler-flux-system unchanged
service/notification-controller unchanged
service/source-controller unchanged
service/webhook-receiver unchanged
deployment.apps/helm-controller configured
deployment.apps/image-automation-controller configured
deployment.apps/image-reflector-controller configured
deployment.apps/kustomize-controller configured
deployment.apps/notification-controller configured
deployment.apps/source-controller configured
networkpolicy.networking.k8s.io/allow-scraping unchanged
networkpolicy.networking.k8s.io/allow-webhooks unchanged
networkpolicy.networking.k8s.io/deny-ingress unchanged
deployment "source-controller" successfully rolled out
```

After a long timee (I think 5 minutes), it prints
```
✗ install failed
```
then quits. It is quite frustrating.

Even with verbose options, the log does not provide any clue on why it get stucks. I would really appreciate some direction on how to proceeed.

Below are some logs that might help:

```
kubectl -n flux-system get pods

NAME                                           READY   STATUS    RESTARTS   AGE
helm-controller-7dbb6cd788-78vr7               1/1     Running   0          17m
image-automation-controller-5dbf64ddc8-5pxjm   1/1     Running   0          17m
image-reflector-controller-778bf765b6-6dnbw    1/1     Running   0          17m
kustomize-controller-756b894785-9ns8h          1/1     Running   0          17m
notification-controller-5979957cb7-bpr79       1/1     Running   0          17m
source-controller-d7d6c79fc-9nbfz              1/1     Running   0          17m
```

```
kubectl get crds | grep fluxcd

alerts.notification.toolkit.fluxcd.io                        2021-01-23T18:40:05Z
buckets.source.toolkit.fluxcd.io                             2021-01-23T18:40:05Z
gitrepositories.source.toolkit.fluxcd.io                     2021-01-23T18:40:05Z
helmcharts.source.toolkit.fluxcd.io                          2021-01-23T18:40:05Z
helmreleases.helm.fluxcd.io                                  2020-10-11T17:51:33Z
helmreleases.helm.toolkit.fluxcd.io                          2021-01-23T18:40:06Z
helmrepositories.source.toolkit.fluxcd.io                    2021-01-23T18:40:06Z
imagepolicies.image.toolkit.fluxcd.io                        2021-01-23T19:32:55Z
imagerepositories.image.toolkit.fluxcd.io                    2021-01-23T19:32:56Z
imageupdateautomations.image.toolkit.fluxcd.io               2021-01-23T19:32:56Z
kustomizations.kustomize.toolkit.fluxcd.io                   2021-01-23T18:40:06Z
providers.notification.toolkit.fluxcd.io                     2021-01-23T18:40:06Z
receivers.notification.toolkit.fluxcd.io                     2021-01-23T18:40:07Z
```

```
flux -v

flux version 0.6.3
```

**Comment by hiddeco** [comment]
Can you try to bootstrap with `--network-policy=false`?

**Comment by binhnguyenduc** [comment]
No luck. Still timeout after `"source-controller" successfully rolled out`.

**Comment by stefanprodan** [comment]
Please try using a different console, what happens is that “kubectl rollout status” hangs.

**Comment by binhnguyenduc** [comment]
Is there anything important after the "rollout status"? What happens if I just ignore that the installation "failed" and proceed as usual?

I tried with different console but the problem persists. Started to think it might have something to do with Rancher or our specific setup.

**Comment by stealthybox** [comment]
Related to this slack thread:
https://cloud-native.slack.com/archives/CLAJ40HV3/p1610978732125700?thread_ts=1610977881.118600&cid=CLAJ40HV3

`flux check` will also hang for the same reason in this case

**Comment by stealthybox** [comment]
@binhnguyenduc your install of Flux is probably fine.
We check to make sure that the controllers are ready, Sources get cloned and become ready, and Kustomizations apply without issue.

Just do a `flux get sources git` and a `flux get kustomizations` to check on those.

**Comment by stealthybox** [comment]
@jonathan-innis is interested in fixing this by replacing the exec's to `kubectl rollout status` to native use of kstatus libs

**Comment by binhnguyenduc** [comment]
A quick update. So I have given up on `bootstrap`. I followed the code in `/cmd` and found out that `bootstrap` will create a `GitRepository` and a `Kustomization` (for the custom path in the git source) after it's done verifying the deployments rollout with `kubectl rollout`.

So I proceed to create those resources manually and Flux v2 seems to be working. 

For anyone who is stucked at the same issue, here is the files I used to finished installation manually:

```
---
apiVersion: source.toolkit.fluxcd.io/v1beta1
kind: GitRepository
metadata:
  name: ***
  namespace: flux-system
spec:
  interval: 1m0s
  ref:
    branch: ***
  secretRef:
    name: ***
  url: ssh://git@***
---
apiVersion: kustomize.toolkit.fluxcd.io/v1beta1
kind: Kustomization
metadata:
  name: ***
  namespace: flux-system
spec:
  interval: 1m
  path: "***"
  prune: true
  sourceRef:
    kind: GitRepository
    name: ***
  validation: client
```

I have to masked the specific details, you will have to provide details for the "***", if you were to use them.

Please let me know if you want me to close this issue.

**Comment by stefanprodan** [comment]
This should be fixed in flux2 v0.7.7, closing this, but feel free to comment here if there are any issues.
